db/db_impl_write.cc

```diff
@@ -148,8 +169,10 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
   // and protects against concurrent loggers and concurrent writes
   // into memtables
 
-  last_batch_group_size_ =
+  uint64_t batch_group_size =
       write_thread_.EnterAsBatchGroupLeader(&w, &write_group);
+  last_batch_group_size_.fetch_add(batch_group_size,
+                                   std::memory_order_relaxed);
```

Comment:
I looked at it more. Since the throttling is done only in commit(), we should only update `last_batch_group_size_` in commit too. Maybe the concurrent code count the same write twice when calculating the write rate, one in prepare and one in commit, and this is wrong. We should only calculate it once. If we only call it in commit(), then we don't need this atomic update. A simply set would be good enough.